packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -0,0 +1,63 @@
+var Db = require('mongodb').Db,
+    MongoClient = require('mongodb').MongoClient,
+    Server = require('mongodb').Server,
+    ObjectID = require('mongodb').ObjectID,
+    Binary = require('mongodb').Binary,
+    GridStore = require('mongodb').GridStore,
+    Grid = require('mongodb').Grid,
+    Code = require('mongodb').Code,
+    assert = require('assert'),
+    crypto = require('crypto');
+
+exports.sourceNodes = (
+  { boundActionCreators, getNode, hasNodeChanged },
+  pluginOptions,
+  done
+) => {
+  const { createNode, deleteNode } = boundActionCreators;
+
+  let serverOptions = pluginOptions.server || { address: 'localhost', port: 27017 };
+
+  let db = new Db(pluginOptions.dbName, new Server(serverOptions.address, serverOptions.port));
+  // Establish connection to db
+  db.open(function(err, db) {
+      if (err) {
+         console.warn(err);
+         return;
+      }
+     
+      let collectionName = pluginOptions.collection || 'documents';
+      let collection = db.collection(collectionName);
+      let cursor = collection.find();
+      // Execute the each command, triggers for each document
+      cursor.each(function(err, item) {
+        console.log(JSON.stringify(item));
+
+        // If the item is null then the cursor is exhausted/empty and closed
+        if(item == null) {  
+          // Let's close the db
+          db.close();
+          done();
+        } else {
+          createNode({
+              // Data for the node.
+              ...item,
+              id: `${item._id}`,
+              parent: `__${collectionName}__`,
+              children: [],
+              collection: collectionName,
+              internal: {
+                  mediaType: 'application/json',
+                  type: `mongoDBDocField`,
```

Comment:
The node type should be based on the collection. Otherwise there could be field type conflicts between fields in different collections e.g. one collection a field is a string and another it's a int.

Also since it looks like you're supporting connecting to multiple DBs which is a cool feature, perhaps have the type name be something like `Mongo${DB_NAME}${COLLECTION_NAME}`.

You might want to make the `collection` field a bit more unique to lessen the chance of conflicting with a field named "collection".